LSPS2/README.md

```diff
@@ -0,0 +1,864 @@
+by means other than BOLT11 invoice, including in a form that
+is not decodable by the payer, but which can be decoded by
+another LN participant.
+
+The only necessary requirement is for the LSP to see an incoming
+HTLC onion that indicates the `jit_channel_scid` as the next hop,
+and as long as the client is able to cause the payer to do this,
+it is would be compatible with this protocol.
+
+> **Non-normative** For example, the client node ID, the
+> `jit_channel_scid`, the LSP node ID, and the
+> `lsp_cltv_expiry_delta` can be encoded as the last hop of
+> a blinded path, for example in a BOLT12 offer or invoice.
+> The payer would not have direct access to the information,
+> but the introduction point of the blinded path would be able
+> to start the decoding of the blinded path.
+
+### 4.  Payment
+
+The client MAY issue the generated invoice to the payer of the
+client, or equivalently inform the payer of the necessary
+information.
+
+The payer can then generate an outgoing payment part to
+deliver the paid amount to the LSP.
+If the client selected "MPP+fixed-invoice" mode, the payer can
+generate multiple outgoing payments parts.
+
+In "MPP+fixed-invoice" mode, the LSP, if it receives a forward
+where the next hop is `jit_channel_scid`:
+
+* SHOULD wait for payment parts of a single payment hash until all
+  parts sum up to at least `payment_size_msat`.
+* MUST NOT fail with `mpp_timeout`, as the final hop is the client
+  and that error is only allowed at the final hop.
+* MUST fail with `unknown_next_peer` if all parts have **NOT**
+  summed up to at least `payment_size_msat` **AND**
+  `opening_fee_params.valid_until` has passed.
+* MAY fail with `unknown_next_peer` if it receives too many payment
+  parts.
+* When all conditions below are true, MUST continue to the next
+  step:
+  * The current time is strictly less than
+    `opening_fee_params.valid_until`.
+  * All received payment parts sum up to at least `payment_size_msat`
+    plus `opening_fee`.
+  * The client is connected.
+
+In "no-MPP+var-invoice" mode, the LSP, if it receives a forward
+where the next hop is `jit_channel_scid`, before
+`opening_fee_params.valid_until`:
+
+* MUST set `payment_size_msat` as the forwarded value of the HTLC.
+* MUST compute the `opening_fee` based on the `payment_size_msat`, and
+  if the computation overflows, MUST fail with `unknown_next_peer`.
+* MUST check that `opening_fee + htlc_minimum_msat < payment_size_msat`,
+  and if that fails, MUST fail with `unknown_next_peer`.
+
+### 5.  Channel Opening And Forwarding
+
+The LSP requests a channel open to the client.
+
+The LSP selects the channel size.
+The LSP MUST ensure the channel is large enough to transmit the
+`payment_size_msat - opening_fee`, plus any reserve needed on the LSP
+side.
+
+The LSP and client MUST negotiate these options and parameters:
+
+* `option_zeroconf` is set.
+* `option_scid_alias` is set.
+* `announce_channel` is `false`.
```

Comment:
When do they negotiate this? Is this during the channel open negotiation? Can we reference the BOLT section?

Reply:
Yes, during channel open.

Reply:
Can we give a hint here that this is done in the regular channel open negotiation? When I read the spec, I was unsure if we need to implement that or if it is already done.